csrc/kernel_cache.cpp

```diff
@@ -534,7 +537,7 @@ std::vector<at::Tensor> FusionExecutorCache::runFusionWithInputs(
   }
 
   if (!kernel_runtime->isCompiled()) {
-    kernel_runtime->compileFusionParallel(args);
+    kernel_runtime->compileFusionParallel(args, auto_schedule_);
```

Comment:
Instead of passing `auto_schedule` everywhere, I think it's more convenient for you and the readers to embed auto_schedule as a field of FusionKernelRuntime (as how you added auto_schedule_ to FusionExecutorCache). getKernelRuntimeFor should be responsible to construct (when cache miss) a FusionKernelRuntime with the right auto_schedule config.

Reply:
done!